packages/fuel-gauge/src/transaction-summary.test.ts

```diff
@@ -621,5 +626,101 @@ describe('TransactionSummary', () => {
         ],
       });
     });
+
+    it('should ensure contract call operations are correctly assembled [WITHDRAW TO BASE LAYER]', async () => {
+      using launched = await launchTestNode();
+
+      const {
+        provider,
+        wallets: [sender],
+      } = launched;
+
+      const recipient = Address.fromB256(
+        '0x00000000000000000000000047ba61eec8e5e65247d717ff236f504cf3b0a263'
+      );
+
+      const tx = await sender.withdrawToBaseLayer(recipient, 10);
+      const result = await tx.waitForResult();
+
+      const { operations } = result;
+
+      expect(operations[0].name).toEqual(OperationName.withdrawFromFuel);
+      expect(operations[0].from?.type).toEqual(1);
+      expect(operations[0].from?.address).toEqual(sender.address.toB256());
+      expect(operations[0].to?.type).toEqual(1);
+      expect(operations[0].to?.address).toEqual(recipient.toB256());
+      expect(operations[0].to?.chain).toEqual(ChainName.ethereum);
+      expect(operations[0].assetsSent).toHaveLength(1);
+      expect(operations[0].assetsSent?.[0].amount).toEqual(bn('10'));
+      expect(operations[0].assetsSent?.[0].assetId).toEqual(provider.getBaseAssetId());
+    });
+
+    it('Should return contract created operations', async () => {
+      using launched = await launchTestNode();
+
+      const {
+        wallets: [wallet],
+      } = launched;
+
+      const tx = await MultiTokenContractFactory.deploy(wallet);
+      const result = await tx.waitForResult();
+
+      expect(result.transactionResult.operations).toHaveLength(1);
+      expect(result.transactionResult.operations[0].name).toEqual(OperationName.contractCreated);
+      expect(result.transactionResult.operations[0].from?.type).toEqual(1);
+      expect(result.transactionResult.operations[0].from?.address).toEqual(wallet.address.toB256());
+      expect(result.transactionResult.operations[0].to?.type).toEqual(0);
+      expect(result.transactionResult.isTypeCreate).toEqual(true);
+    });
+
+    it('should have no assets returned for contract call operations', async () => {
+      using launched = await launchTestNode({
+        contractsConfigs: [
+          {
+            factory: TokenContractFactory,
```

Comment:
@maschad Sorry for the lack of context but I did not understand why we need this validation

Reply:
I was trying to find a way to replicate [this test in operations](https://github.com/FuelLabs/fuels-ts/blob/mc/chore/add-integration-tests-operations/packages/account/src/providers/transaction-summary/operations.test.ts#L366) the most immediate e2e use case was to to do a contract call with no assets sent in the call.